### PR TITLE
add title_tag param to tiles

### PR DESCRIPTION
**Pull request by adipasquale**
Some DSFR components use titles, such as the `Tile` which uses a `h4`. The level of this heading seems arbitrary in the DSFR examples and does not impact its aspect. However it’s a problem for accessibility : heading levels are supposed to be coherent, you should not follow an `h1` with an `h4` for example. cf [this aXe rule](https://dequeuniversity.com/rules/axe/4.6/heading-order?application=axeAPI) and [RGAA 9.1](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#9.1).

In this PR I introduce a `title_tag` param specific to the `TileComponent`. The pattern will probably be reused in other components, so it makes sense to take some time to consider this param name. I thought about `title_level` or `heading_level` but I feel it’s more explicit to pass `:h3` than `3` as an argument. Maybe `heading_tag` would be a better name though. 

There is a small incoherence with the `_tag` suffix : in the templates we have to call `content_tag(title_tag, "blah")` which makes little sense : `content_tag` is a HTML helper whereas `title_tag` will only return `h2`.

@freesteph if you have smart ideas I'm all ears :) 

Screenshot 2023-02-17 at 17 44 51

this will unblock https://github.com/betagouv/collectif-objets/pull/706

cf https://mattermost.incubateur.net/betagouv/pl/4pxd4u6gwbb5per8pb153yezhe

### CODE REVIEW

File: app/components/dsfr_component/base.rb [hunk 1]
```diff
@@ -5,6 +5,8 @@ class DsfrComponent::Base < ViewComponent::Base
 
   delegate :config, to: Dsfr::Components
 
+  TITLE_TAGS = %i[h1 h2 h3 h4 h5 h6].freeze
```
Comment (freesteph) [inline app/components/dsfr_component/base.rb:8]: I don't think we need to enforce the header level. It should be a number, and that's about it.

File: app/components/dsfr_component/tile_component.rb [hunk 2]
```diff
@@ -6,21 +6,24 @@ class TileComponent < DsfrComponent::Base
     # @param image_alt [String] L'alternative de l'image. Doit a priori être vide si l'image est illustrative et n'apporte pas de valeur sémantique.
     # @param description [String] description (optional)
     # @param orientation [String] :horizontal or :vertical
-    def initialize(title:, url:, image_src: nil, image_alt: "", description: nil, orientation: :vertical, classes: [], html_attributes: {})
+    # @param title_tag [Symbol] :h1, :h2, :h3, :h4 (default), :h5, :h6
```
Comment (freesteph) [inline app/components/dsfr_component/tile_component.rb:9]: I prefer your argument of using `heading_level: number`. It's the easiest we can make the interface, and it keeps more semantic on our side which I think is nice.

File: guide/content/components/tile.haml [hunk 3]
```diff
@@ -23,5 +23,10 @@ title: Tuile (Tile)
 = render('/partials/example.haml', caption: "Tuile horizontale", code: tile_horizontal) do
   Tuile horizontale avec image, titre et description
 
+= render('/partials/example.haml', caption: "Tuile avec un niveau de titre spécifique", code: tile_title_tag) do
+  L’argument `title_tag` ne change pas l’aspect de la tuile mais permet d’adapter le niveau du heading à votre page pour améliorer l’accessibilité en respectant la cohérence de niveaux de titres.
```
Comment (freesteph) [inline guide/content/components/tile.haml:27]: This should be enclosed as `:markdown`

**Review by freesteph — COMMENTED** [review]
I prefer the idea of `heading_level: number`, would you mind amending that way? Otherwise looks fine.

File: guide/content/components/tile.haml (hunk 3 above)
Comment (adipasquale) [reply guide/content/components/tile.haml:27]: good catch, thanks

File: app/components/dsfr_component/tile_component.rb (hunk 2 above)
Comment (adipasquale) [reply app/components/dsfr_component/tile_component.rb:9]: fine with me, I don’t have a strong opinion on this, I'll update it

File: app/components/dsfr_component/base.rb (hunk 1 above)
Comment (adipasquale) [reply app/components/dsfr_component/base.rb:8]: fine with me but there are indeed only 6 levels cf https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements

File: app/components/dsfr_component/base.rb [hunk 4]
```diff
@@ -5,6 +5,8 @@ class DsfrComponent::Base < ViewComponent::Base
 
   delegate :config, to: Dsfr::Components
 
+  HEADING_LEVELS = [1, 2, 3, 4, 5, 6].freeze
```
Comment (freesteph) [inline app/components/dsfr_component/base.rb:8]: This is technically a range but I'll sleep the same tonight

**Review by freesteph — APPROVED** [review]

**Pull request merged**